### PR TITLE
os: Add support for macOS (Take 2)

### PR DESCRIPTION
**Pull request by dylanaraps**
This PR adds support for macOS (iTerm2 only).

Can those with access to a macOS machine with iTerm2 test this PR for me? 

TODO:

- [ ] Add wallpaper support.
    - [ ] `osascript` command for older versions of macOS.
    - [ ] `sqlite3` command for newer versions of macOS.
- [x] Verify that the escape sequences work.

**Those of you using macOS and would like to help test:**

Do any of these commands change iTerm2's background color to `#FFFFFF`?

```sh
printf "%b\n" "\033]\a1337;SetColors=bg=FFFFFF"
printf "%b\n" "\033]PhFFFFFF\033\\"
```

Sources: 

- https://www.iterm2.com/documentation-escape-codes.html
- https://github.com/gnachman/iTerm2/blob/master/tests/setpal
- https://gitlab.com/gnachman/iterm2/issues/454
- https://github.com/gnachman/iTerm2/blob/master/tests/set_colors.applescript
- https://github.com/gnachman/iTerm2/blob/master/tests/it2setcolor

### CODE REVIEW

**Comment by dylanaraps** [comment]
Note: For those who read this.

Support for macOS is possible (iTerm2 only), the only issue is that I can't get a macOS machine I can test on. What I need to test is iTerm2's proprietary escape sequences. 

iTerm2 supports a similar sequence to the Linux terminals to change colors but it works differently and the reference page in the docs isn't very clear. 

Here's the page (Scroll down to "Change the color palette"): 

> https://www.iterm2.com/documentation-escape-codes.html

> Change the color palette
>
> `^[]Pnrrggbb^[\`
>
> Replace "n" with:
>
>    `0-f (hex) = ansi color`
    `g = foreground`
    `h = background`
    `i = bold color`
    `j = selection color`
    `k = selected text color`
   ` l = cursor`
    `m = cursor text`
>
> rr, gg, bb are 2-digit hex value (for example, "ff"). Example in bash that changes the foreground color blue:
>
> `echo -e "\033]Pg4040ff\033\\"`

Basically what we need to figure out is the exact sequences we need to modify `background`, `foreground`, `cursor` and colors `0-15`. If anyone with a macOS system would like to test I'd be grateful as I can't find these sequences in use anywhere as a reference.

More info:

- https://gitlab.com/gnachman/iterm2/issues/454

**Comment by dylanaraps** [comment]
This is another alternative: https://github.com/gnachman/iTerm2/blob/master/tests/set_colors.applescript

I'm not sure about how well it works but it's something we can try out if we fail to get the sequences working.

**Comment by dylanaraps** [comment]
Here's a third alternative: https://github.com/gnachman/iTerm2/blob/master/tests/it2setcolor

**Comment by aeikenberry** [comment]
`printf "%b\n" "\033]PhFFFFFF\033\\"` worked for me. Changed background white.
`printf "%b\n" "\033]\a1337;SetColors=bg=FFFFFF"` didn't seem to have any effect.

**Comment by dylanaraps** [comment]
Can you try out the PR as a whole and let me know if the entire scheme is set correctly? 

Here are the steps to test it out:

1. `git clone https://github.com/dylanaraps/pywal`
2. `cd pywal`
3. `git checkout macos2`
4. `python3 -m pywal -i /path/to/img.jpg`

**Comment by aeikenberry** [comment]
Yeah, it definitely works, the only thing that seems off is that the background color is consistently set the same as the foreground. Do you need any other details? example pic and generated scheme?

**Comment by dylanaraps** [comment]
Can I give you a test image so we can compare results between my machine and yours? Just run `wal` using the wallpaper below and this one line script in the terminal.

```sh
# Test the colors.
for x in 0 1 4 5 7 8; do for i in {30..37}; do for a in {40..47}; do echo -ne "\e[$x;$i;$a""m\\\e[$x;$i;$a""m\e[0;37;40m "; done; echo; done; done; echo "";
```



My machine:

**Comment by aeikenberry** [comment]
screenshot 2017-07-30 23 39 05

**Comment by dylanaraps** [comment]
Awesome, thanks for testing!

I've fixed the issue, please do a git pull and let me know if it works now. :)

(You can add `-c` to the end of the `wal` command to make it regenerate the scheme.)

**Comment by aeikenberry** [comment]
Fix looks good! Thanks for all the good work!

**Comment by dylanaraps** [comment]
Thanks for testing! 

I'll clean this up a little and then release a new version you can `pip install`.
